From a0f788c981dd733a811e057b88c6dc1f99b9d791 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Tue, 31 Oct 2023 15:15:46 +0000
Subject: [PATCH] libstdc++: fix incomplete type in blink::LocalFrame and
 client

Full definition of blink::WebBackgroundResourceFetchAssets is
required for scoped_refptr release.

Bug: 957519
Change-Id: If69e3c575f48132db6d74ce8279fc63b21f79125
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4994625
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Reviewed-by: David Bokan <bokan@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1217593}
---
 third_party/blink/renderer/core/frame/local_frame.h        | 2 +-
 third_party/blink/renderer/core/frame/local_frame_client.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/core/frame/local_frame.h b/third_party/blink/renderer/core/frame/local_frame.h
index b233430e604976..7efb913cb4ec55 100644
--- a/third_party/blink/renderer/core/frame/local_frame.h
+++ b/third_party/blink/renderer/core/frame/local_frame.h
@@ -67,6 +67,7 @@
 #include "third_party/blink/public/mojom/script/script_evaluation_params.mojom-blink-forward.h"
 #include "third_party/blink/public/mojom/use_counter/metrics/web_feature.mojom-blink-forward.h"
 #include "third_party/blink/public/platform/task_type.h"
+#include "third_party/blink/public/platform/web_background_resource_fetch_assets.h"
 #include "third_party/blink/public/platform/web_vector.h"
 #include "third_party/blink/public/web/web_script_execution_callback.h"
 #include "third_party/blink/renderer/core/core_export.h"
@@ -157,7 +158,6 @@ class SystemClipboard;
 class TextFragmentHandler;
 class TextSuggestionController;
 class VirtualKeyboardOverlayChangedObserver;
-class WebBackgroundResourceFetchAssets;
 class WebContentSettingsClient;
 class WebInputEventAttribution;
 class WebPluginContainerImpl;
diff --git a/third_party/blink/renderer/core/frame/local_frame_client.h b/third_party/blink/renderer/core/frame/local_frame_client.h
index b83ee768ac5622..c0fa11ea9ec9f3 100644
--- a/third_party/blink/renderer/core/frame/local_frame_client.h
+++ b/third_party/blink/renderer/core/frame/local_frame_client.h
@@ -56,6 +56,7 @@
 #include "third_party/blink/public/mojom/portal/portal.mojom-blink-forward.h"
 #include "third_party/blink/public/platform/child_url_loader_factory_bundle.h"
 #include "third_party/blink/public/platform/scheduler/web_scoped_virtual_time_pauser.h"
+#include "third_party/blink/public/platform/web_background_resource_fetch_assets.h"
 #include "third_party/blink/public/platform/web_content_settings_client.h"
 #include "third_party/blink/public/platform/web_effective_connection_type.h"
 #include "third_party/blink/public/platform/web_worker_fetch_context.h"
@@ -104,7 +105,6 @@ class ResourceError;
 class ResourceRequest;
 class ResourceResponse;
 class SourceLocation;
-class WebBackgroundResourceFetchAssets;
 class WebContentCaptureClient;
 class WebDedicatedWorkerHostFactoryClient;
 class WebLocalFrame;
